api/src/main/java/org/openmrs/module/appointments/service/impl/SMSService.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.bahmni.webclients.ClientCookies;
+import org.openmrs.Location;
+import org.openmrs.LocationTag;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appointments.connection.OpenmrsLogin;
+import org.openmrs.module.appointments.model.SMSRequest;
+
+import java.text.MessageFormat;
+import java.util.*;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import static org.openmrs.module.appointments.util.DateUtil.convertUTCToGivenFormat;
+
+public class SMSService {
+
+    private final static String APPOINMTMENT_REMINDER_SMS_TEMPLATE = "sms.appointmentReminderSMSTemplate";
+    private final static String APPOINMTMENT_REMINDER_SMS_TEMPLATE_NO_PROVIDER = "sms.appointmentReminderSMSTemplateWithoutProvider";
+    private final static String SMS_TIMEZONE = "bahmni.sms.timezone";
+    private final static String SMS_DATEFORMAT = "bahmni.sms.dateformat";
+    private final static String SMS_URL = "bahmni.sms.url";
+    private static Logger logger = LogManager.getLogger(SMSService.class);
+    private OpenmrsLogin openmrsLogin;
+    private AppointmentVisitLocation appointmentVisitLocation;
+
+    public void setOpenmrsLogin(OpenmrsLogin openmrsLogin) {
+        this.openmrsLogin = openmrsLogin;
+    }
+
+    public void setAppointmentVisitLocation(AppointmentVisitLocation appointmentVisitLocation) {
+        this.appointmentVisitLocation = appointmentVisitLocation;
+    }
+
+    public String getAppointmentMessage(String name, String familyName, String id, Date appointmentDate, String service, List<String> providers, Location location) {
+        String smsTemplate = Context.getAdministrationService().getGlobalProperty(APPOINMTMENT_REMINDER_SMS_TEMPLATE);
+        String smsTemplateNoProvider = Context.getAdministrationService().getGlobalProperty(APPOINMTMENT_REMINDER_SMS_TEMPLATE_NO_PROVIDER);
+        String smsTimeZone = Context.getMessageSourceService().getMessage(SMS_TIMEZONE, null, new Locale("en"));
+        String smsDateFormat = Context.getMessageSourceService().getMessage(SMS_DATEFORMAT, null, new Locale("en"));
+        String date =convertUTCToGivenFormat(appointmentDate, smsDateFormat, smsTimeZone);
+        String helpdeskNumber = Context.getAdministrationService().getGlobalPropertyObject("clinic.helpDeskNumber").getPropertyValue();
+        LocationTag abc = Context.getLocationService().getLocationTagByName("Visit Location");
+        List<Location> locations = Context.getLocationService().getLocationsHavingAnyTag(Collections.singletonList(abc));
+        String clinicName = (abc != null && !locations.isEmpty()) ? locations.get(0).getName() : "xxxxx";
```

Comment:
Can we call this facilityName instead of clinicName?

Reply:
Yah will do the change